### PR TITLE
Message system refactor -- login

### PR DESCRIPTION
**Pull request by Pacane**

### CODE REVIEW

File: server/lib/messages/handlers/kick_handler.dart [hunk 1]
```diff
@@ -1,4 +1,4 @@
-library kick_event_handler;
+library kick_handler_handler;
```
Comment (meriouma) [inline server/lib/messages/handlers/kick_handler.dart]: kick_handler_handler?

File: shared/lib/messages/message_factory.dart [hunk 2]
```diff
@@ -25,6 +26,8 @@ class MessageFactory {
         return new ErrorEvent.fromJson(content);
       case CardSelectionEvent.MSG_TYPE:
         return new CardSelectionEvent.fromJson(content);
+      case LoginEvent.MSG_TYPE:
+        return new LoginEvent.fromJson(content);
       default:
         return null;
```
Comment (meriouma) [inline shared/lib/messages/message_factory.dart]: Maybe we could add a log call in the `default` case

**Comment by meriouma** [comment]
Nice, LGTM

File: client/lib/components/game/game.dart [hunk 3]
```diff
@@ -157,15 +158,7 @@ class GameComponent implements ScopeAware, AttachAware, DetachAware {
...
-    var loginInfo = {
-        'login' :
-        {
-            'gameId': currentGame.getGameId(),
-            'username': currentUser.userName
-        }
-    };
-
-    socketCommunication.sendSocketMsg(loginInfo);
+    socketCommunication.sendSocketMsg(new LoginEvent(currentGame.getGameId(), currentUser.userName));
```
Comment (meriouma) [inline client/lib/components/game/game.dart]: You didn't have to call `.toJson()`? I think I had to for the KickEvent

File: server/lib/messages/handlers/login_handler.dart [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import 'package:logging/logging.dart';
+
+import 'package:poker_planning_server/broadcaster.dart';
+import 'package:poker_planning_server/repository/game_repository.dart';
+
+import 'package:poker_planning_shared/game.dart';
+import 'package:poker_planning_shared/messages/handlers/connection_message_handler.dart';
+import 'package:poker_planning_shared/messages/login_event.dart';
+
```
Comment (christiangoudreau) [inline server/lib/messages/handlers/login_handler.dart]: rm

File: shared/lib/messages/message_factory.dart (hunk 2 above)
Comment (Pacane) [inline shared/lib/messages/message_factory.dart]: Yes probably.

File: client/lib/components/game/game.dart (hunk 3 above)
Comment (Pacane) [inline client/lib/components/game/game.dart]: No in fact, when we `JSON.encode()` an object, it creates a default Encoder that uses `toJson()`
see : 
```
  /**
   * Converts [value] to a JSON string.
   *
   * If value contains objects that are not directly encodable to a JSON
   * string (a value that is not a number, boolean, string, null, list or a map
   * with string keys), the [toEncodable] function is used to convert it to an
   * object that must be directly encodable.
   *
   * If [toEncodable] is omitted, it defaults to a function that returns the
   * result of calling `.toJson()` on the unencodable object.
   */
String encode(Object value, {toEncodable(var object)}) {
    if (toEncodable == null) toEncodable = _toEncodable;
    if (toEncodable == null) return encoder.convert(value);
    return new JsonEncoder(toEncodable).convert(value);
  }
```

Comment (meriouma) [inline client/lib/components/game/game.dart]: Good to know, then we could probably remove the .toJson on the kick event

**Pull request merged**